test/integration/agent/npl_test.go

```diff
@@ -0,0 +1,155 @@
+//go:build linux
+// +build linux
+
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package agent
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	nplcontroller "antrea.io/antrea/pkg/agent/nodeportlocal/k8s"
+	portcache "antrea.io/antrea/pkg/agent/nodeportlocal/portcache"
+	rules "antrea.io/antrea/pkg/agent/nodeportlocal/rules"
+	iptables "antrea.io/antrea/pkg/agent/util/iptables"
+)
+
+func IntTest(t *testing.T) {
+	portTable, err := portcache.NewPortTable(61000, 62000)
+	if err != nil {
+		t.Fatalf("Failed to initialize porttable: %v", err)
+	}
+	nplCltr := nplcontroller.NewTestingNPLController(portTable)
+	stopCh := make(chan struct{})
+	nplcontroller.SyncRuleInterval = 2
+	nplCltr.SyncRules(stopCh)
+	time.Sleep(nplcontroller.SyncRuleInterval + 1*time.Second)
+	close(stopCh)
+}
+
+func TestNPLIptablesRestore(t *testing.T) {
+	portTable, err := portcache.NewPortTable(61000, 62000)
+	if err != nil {
+		t.Fatalf("Failed to initialize porttable: %v", err)
+	}
+	// get testing NPLController
+	synced := make(chan struct{})
+	nplCltr := nplcontroller.NewTestingNPLController(portTable)
```

Comment:
I think nplCltr should be renamed to nplCtrl. It seems Ctrl is often used as short form of Control.